### PR TITLE
providers lock tests

### PR DESCRIPTION
**Pull request by mildwonkey**
There are a few small changes in this PR:

- Tests for providers lock, go me
- I replaced a clearly copy-pasta'd error message with a more appropriate "Invalid provider argument" error - please do nitpick my wording.
- Writing tests for providers lock unearthed a very tiny bug in `getproviders.ParsePlatform`, so a fix for that (including tests) is in here too. 
- Barely worth mentioning providers mirror "tests" (emphasis on the airquotes)

### CODE REVIEW

File: internal/getproviders/types.go [hunk 1]
```diff
@@ -137,6 +137,11 @@ func (p Platform) LessThan(other Platform) bool {
 // ParsePlatform parses a string representation of a platform, like
 // "linux_amd64", or returns an error if the string is not valid.
 func ParsePlatform(str string) (Platform, error) {
+	parts := strings.Split(str, "_")
+	if len(parts) != 2 {
+		return Platform{}, fmt.Errorf("must be two words separated by an underscore")
+	}
+
```
Comment (jbardin) [inline internal/getproviders/types.go:144]: I think this check replaces the next one

File: .gitignore [hunk 2]
```diff
@@ -4,6 +4,7 @@
 example.tf
 terraform.tfplan
 terraform.tfstate
+.terraform.lock.hcl
```
Comment (jbardin) [inline .gitignore:7]: is this from a test that can be cleaned up? I'd actually like to remove files from gitignore that can alter behavior, as we've had many tests over the years that fail due to ignored files, which of course you can't track down using git.

File: internal/getproviders/types.go (hunk 1 above)
Comment (mildwonkey) [reply internal/getproviders/types.go:144]: it does not, that's why I added it  and the supporting tests -- those and the command test I wrote failed, initially.

Comment (mildwonkey) [reply internal/getproviders/types.go:144]: It only checks that there is _an_ underscore, so you can have multiples.

Comment (jbardin) [reply internal/getproviders/types.go:144]: Sorry, I mean your new one supersedes the following check, which manually splits the part in 2. Deleting the next block and using `os, arch := parts[0], parts[1]` should work exactly the same.

Comment (mildwonkey) [reply internal/getproviders/types.go:144]: That's true. I almost made that change, but decided to leave the rest  as-is. I'll go do that instead. Thanks!

File: .gitignore (hunk 2 above)
Comment (mildwonkey) [reply .gitignore:7]: No, not a test, but excellent point; an accidental cli command (wrong directory) is what prompted adding this. I can remove it, I was just acting on the observation that we've added many other "standard" terraform artifacts in our .gitignore. 🤷

Comment (mildwonkey) [reply .gitignore:7]: (sometimes I get _obsessively_ consistent, which isn't always helpful!)

**Review by jbardin — APPROVED** [review]

**Pull request merged**